source/lac/sparse_direct.cc

```diff
@@ -1344,6 +1351,9 @@ SparseDirectMUMPS::vmult(VectorType &dst, const VectorType &src) const
             &local_array);
 #  endif
         }
+      else
+        Assert(false, ExcInternalError());
```

Comment:
Would you mind changing this to `DEAL_II_ASSERT_UNREACHABLE();`? I think that more clearly represents what we want.